lib/streamlit/ConfigOption.py

```diff
@@ -138,7 +138,9 @@ def __init__(
         """
         # Parse out the section and name.
         self.key = key
-        key_format = r"(?P<section>\_?[a-z][a-z0-9]*)\.(?P<name>[a-z][a-zA-Z0-9]*)$"
+        key_format = (
+            r"(?P<section>\_?[a-z][a-z0-9]*)\.(?P<name>[v_0-9]*[a-z][a-zA-Z0-9]*)$"
```

Comment:
I usually find regex'es like this hard to read -- maybe an example of what this would match could help?

Reply:
good idea and i should also update the tests with more cases

Reply:
You can also add comments:

```
my_regex = (
   # Capture a group called "section"
    "(?<section>"
        # Match anything starting with "foo", up to a space.
        foo[^ ]*"
    ")"
    # Match the version number.
    "v[0-9]+"
)
```

And I always wondered about whether [literate regex libraries like this one](https://github.com/mtrencseni/rxe) are useful in real life. But that's a whole other discussion!

Reply:
BTW, nice use of named capture groups. We should use those _everywhere_!